@here/harp-text-canvas/lib/typesetting/SmoothPathTypesetter.ts

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as THREE from "three";
+
+import { MathUtils } from "@here/harp-utils";
+import { GlyphData } from "../rendering/GlyphData";
+import { QUAD_VERTEX_MEMORY_FOOTPRINT } from "../rendering/TextGeometry";
+import { FontStyle, FontVariant } from "../rendering/TextStyle";
+import { TypesettingUtils } from "../utils/TypesettingUtils";
+import { UnicodeUtils } from "../utils/UnicodeUtils";
+import { PathTypesetter, PathTypesettingParameters } from "./PathTypesetter";
+
+/**
+ * @hidden
+ * [[Typesetter]] implementation that arranges glyphs alongside a specified path.
+ */
+export class SmoothPathTypesetter extends PathTypesetter {
+    private m_tempGlyphs: GlyphData[] = [];
+    private m_tempPoints: THREE.Vector2[] = [];
+    private m_tempAngles: number[] = [];
+    private m_tempAngleDelta: number[] = [];
+    private m_tempMirrored: boolean[] = [];
+
+    constructor() {
+        super();
+        this.reset();
+    }
+
+    arrangeGlyphs(params: PathTypesettingParameters): boolean {
+        if (super.arrangeGlyphs(params) === false) {
+            return false;
+        }
+
+        const result = this.commitLine();
+        this.reset();
```

Comment:
This adds a bit of GC pressure in a performance critical part of the draw loop. Would be interesting to compare how much the impact is in real scenes.

Reply:
Do you think, emptying existing arrays would help? With splice or so.